### PR TITLE
Pass hierarchy depth level to custom fields (#135)

### PR DESCRIPTION
**Pull request by ai-ba**
See #135

### CODE REVIEW

File: src/components/HierarchyTable/HierarchyTable.tsx [hunk 1]
```diff
@@ -232,6 +232,7 @@ export const HierarchyTable: FunctionComponent<HierarchyTableProps> = (props) =>
                         widgetName={props.meta.name}
                         widgetFieldMeta={item}
                         readonly
+                        customProps={{hierarchyDepth: indentLevel + 1}}
```
Comment (Dergash) [inline src/components/HierarchyTable/HierarchyTable.tsx]: This should memoized or `<Field />` will be recreated every render

File: src/components/Field/Field.tsx [hunk 2]
```diff
@@ -29,6 +29,11 @@ import {CustomizationContext} from '../../components/View/View'
 import {InteractiveInput} from '../../components/ui/InteractiveInput/InteractiveInput'
 import HistoryField from '../../components/ui/HistoryField/HistoryField'
 
+export interface FieldCustomProps {
+    hierarchyDepth?: number,
```
Comment (Dergash) [inline src/components/Field/Field.tsx]: This way `Field` still knows about Hierarchy. Custom properties should be of any type and the actual typing for the specific props should be declared in the component that utilizes those props

File: src/components/HierarchyTable/HierarchyTable.tsx (hunk 1 above)
Comment (ai-ba) [reply src/components/HierarchyTable/HierarchyTable.tsx]: Fixed.

File: src/components/Field/Field.tsx (hunk 2 above)
Comment (ai-ba) [reply src/components/Field/Field.tsx]: Fixed.

**Pull request merged**